coderd/notifications/dispatch/inbox.go

```diff
@@ -0,0 +1,81 @@
+package dispatch
+
+import (
+	"context"
+	"encoding/json"
+	"text/template"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+
+	"github.com/google/uuid"
+
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/database/dbtime"
+	"github.com/coder/coder/v2/coderd/notifications/types"
+	markdown "github.com/coder/coder/v2/coderd/render"
+)
+
+type inboxStore interface {
+	InsertInboxNotification(ctx context.Context, arg database.InsertInboxNotificationParams) (database.InboxNotification, error)
+}
+
+// InboxHandler is responsible for dispatching notification messages in the Coder Inbox.
+type InboxHandler struct {
+	log   slog.Logger
+	store inboxStore
+}
+
+func NewInboxHandler(log slog.Logger, store inboxStore) *InboxHandler {
```

Comment:
You're exporting `NewInboxHandler` but you're accepting a `store` arg with an unexported interface.